src/benchmarks/micro/libraries/System.Diagnostics/Perf_Activity.cs

```diff
@@ -0,0 +1,373 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace System.Diagnostics
+{
+    [MemoryDiagnoser]
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Perf_Activity
+    {
+        private static readonly ActivitySource s_ActivitySource;
+        private static readonly ActivityListener s_ActivityListener;
+        private static readonly Activity s_Activity;
+        private static readonly ActivityLink s_ActivityLink;
+
+        static Perf_Activity()
+        {
+            s_ActivitySource = new ActivitySource("TestActivitySource");
+
+            s_ActivityListener = new ActivityListener
+            {
+                ShouldListenTo = s => s.Name == "TestActivitySource",
+
+                GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> o) => ActivityDataRequest.AllDataAndRecorded
+            };
+
+            ActivitySource.AddActivityListener(s_ActivityListener);
+
+            s_Activity = s_ActivitySource.StartActivity(
+                "TestActivity",
+                ActivityKind.Internal,
+                parentContext: default,
+                tags: new Dictionary<string, object>
+                {
+                    ["tag1"] = "string1",
+                    ["tag2"] = 1,
+                    ["tag3"] = "string2",
+                    ["tag4"] = false,
+                },
+                links: new ActivityLink[]
+                {
+                    new ActivityLink(default),
+                    new ActivityLink(default),
+                    new ActivityLink(default),
+                    new ActivityLink(default),
+                });
+
+            s_Activity.AddEvent(new ActivityEvent("TestEvent1"));
+            s_Activity.AddEvent(new ActivityEvent("TestEvent2"));
+            s_Activity.AddEvent(new ActivityEvent("TestEvent3"));
+            s_Activity.AddEvent(new ActivityEvent("TestEvent4"));
+
+            s_Activity.Stop();
+
+            s_ActivityLink = new ActivityLink(
+                default,
+                new ActivityTagsCollection(
+                    new Dictionary<string, object>
+                    {
+                        ["tag1"] = "string1",
+                        ["tag2"] = 1,
+                        ["tag3"] = "string2",
+                        ["tag4"] = false,
+                    }));
+        }
+
+        [Params(5000)]
+        public int NumberOfActivities { get; set; }
+
+        [Benchmark]
+        public void EnumerateActivityTags()
+        {
+            int total = 0;
+
+            for (int i = 0; i < NumberOfActivities; i++)
```

Comment:
>for [](start = 12, length = 3)

do we really need to loop? Benchmark already repeat the method execution anyway and we can get he exact number for the enumeration. 

I would suggest too to create multiple static activities with different tags count. we can have the perf numbers for short, medium and long tags count.

Reply:
Removed the looping. Added small/large tests.